### ISSUE TITLE
Bug: c# Binary response truncated for start-api

### ISSUE THREAD

**Issue by sudsy** [aws/aws-sam-cli#4296]
### Description:
When a stream contains a certain sequence of bytes, the response stream is truncated by sam local. This behaviour is inconsistent with the same function deployed to AWS. I noticed that despite the response being a binary stream sam local attaches the Content-Type': 'application/json' header while the same function deployed to AWS has the correct header 'Content-Type': 'application/octet-stream' making me think this is related to #325

Note that it is only a specific byte sequence that causes this. If I write zeroes to the stream they are not truncated. I am guessing that it is some sort of preamble that is being detected and stripped. Sadly the content that I want to send (A serialised protobuf message) starts with this sequence.

### Steps to reproduce:
The following Lambda function written in c# deployed with Runtime.DOTNET_6 using CDK demonstrates the issue:

 ```
public Stream FunctionHandler(Stream inputStream, ILambdaContext context)
    {
        var ms = new MemoryStream();
        ms.WriteByte(0x12);
        ms.WriteByte(0x10);
        ms.WriteByte(0x0A);
        ms.WriteByte(0x0B);
        ms.Position = 0;
        return ms;
    }
```

### Observed result:
aws lambda invoke --endpoint-url http://127.0.0.1:3001 ....

shows that the response is empty

2022-10-12 00:26:39,497 - MainThread - botocore.parsers - DEBUG - Response headers: {'Content-Type': 'application/json', 'Content-Length': '0', 'Server': 'Werkzeug/1.0.1 Python/3.7.10', 'Date': 'Wed, 12 Oct 2022 00:26:39 GMT'}

while the same request to aws looks like this:

2022-10-12 00:26:31,180 - MainThread - botocore.parsers - DEBUG - Response headers: {'Date': 'Wed, 12 Oct 2022 00:26:32 GMT', 'Content-Type': 'application/octet-stream', 'Content-Length': '4', 'Connection': 'keep-alive', 'x-amzn-RequestId': 'd15ffe5c-e6e8-4f45-8f67-ef7afdc67fa9', 'x-amzn-Remapped-Content-Length': '0', 'X-Amz-Executed-Version': '$LATEST', 'X-Amzn-Trace-Id': 'root=1-634609b8-53817f9f61a9b43f2b67a41e;sampled=0'}

### Expected result:

I was expecting Sam local to behave the same as AWS

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. MacOS vscode development container mcr.microsoft.com/vscode/devcontainers/dotnet:6.0 running on docker desktop 
2. `sam 1.58.0 running using docker in docker from development container `:
3. AWS region: us-west-2

**Comment by lucashuy** [comment]
Hi thanks for raising this issue. We'll have to take a look at this and get back to you with any next steps

**Comment by mndeveci** [comment]
Thanks for raising this issue.

I tried to debug the `sam local start-lambda` call with following example;
```
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using System.Threading.Tasks;
using System.IO;

using Amazon.Lambda.APIGatewayEvents;
using Amazon.Lambda.Core;

// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.SystemTextJson.DefaultLambdaJsonSerializer))]

namespace HelloWorld
{

    public class Function
    {


        public Stream FunctionHandler(Stream inputStream, ILambdaContext context)
        {
                var ms = new MemoryStream();
                ms.WriteByte(0x12);
                ms.WriteByte(0x10);
                ms.WriteByte(0x0A);
                ms.WriteByte(0x0B);
                //ms.WriteByte(0x30);
                ms.Position = 0;
                return ms;
         }
    }
}
```

Once the function is invoked, this method handles the response of it: https://github.com/aws/aws-sam-cli/blob/develop/samcli/local/services/base_local_service.py#L115

As you can see, the code splits the response with a new line feed (`0x0A`) and assumes everything above is the lambda logs, and everything below is the actual lambda response. And once it does the split, it also strips the string, in this case gets rid of the `0x30`. So in your example, AWS CLI will get nothing from the lambda invocation.

If you try with a lambda function which writes bytes other than new line, this just works since it doesn't enter the if block there.

I will be checking with the team why we needed that if statement and if we can fix it for this use case.

**Comment by mndeveci** [comment]
Hi there,

We confirmed that this is a bug, and raised a PR to address the issue.

We are going to resolve this issue once it is merged and released.

Thanks again for reporting this!

**Comment by mndeveci** [comment]
Fix is released with [v1.71.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.71.0)
